modules/collect/src/test/java/com/opengamma/strata/collect/MapStreamTest.java

```diff
@@ -141,6 +152,14 @@ public void mapKeysAndValuesToValues() {
     assertThat(result).isEqualTo(expected);
   }
 
+  @Test
+  public void mapBoth() {
+    Map<String, Short> expected =
+        ImmutableMap.of("one - 1", (short) 0, "two - 1", (short) 1, "three - 1", (short) 2, "four - 1", (short) 3);
+    Map<String, Short> result = MapStream.of(map).mapBoth((k, v) -> entry(k + " - 1", (short) (v - 1))).toMap();
```

Comment:
Why all the (short) casting?

Reply:
To demonstrate that the type of the result can be changed

Reply:
What type is being changed? You're going from `Map<String, Short> -> Map<String, Short>`

Reply:
You can write `short s = 1` to convert the literal `1`. But you cannot convert that way to a capital S `Short` - the `1` is interpreted as an `int`, hence the cast.

Reply:
Ahah, got you. Isn't that due to `Guavate.entry` doing the boxing for you?